shell/platform/windows/flutter_windows_engine.cc

```diff
@@ -830,4 +984,56 @@ void FlutterWindowsEngine::OnChannelUpdate(std::string name, bool listening) {
   }
 }
 
+// Platform view methods:
+void FlutterWindowsEngine::RegisterPlatformViewType(
+    const std::string& type,
+    PlatformViewCreationContext factory) {
+  platform_view_types_.insert({type, factory});
+}
+
+void FlutterWindowsEngine::CreatePlatformView(PlatformViewCreationParams args) {
+  FML_LOG(ERROR) << "Creating a platform view of type \"" << args.view_type
+                 << "\"";
+  const auto itr = platform_view_types_.find(std::string(args.view_type));
+  if (itr == platform_view_types_.end()) {
+    FML_LOG(ERROR) << "View type not registered";
+    return;
+  }
+  args.user_data = itr->second.user_data;
+  pending_platform_views_.insert({args.id, {itr->second.factory, args}});
+  /*task_runner_->PostTask([this, itr, args]() {
+    HWND platform_view = itr->second(&args);
+    platform_views_.insert({args.id, platform_view});
+  });*/
+}
+
+void FlutterWindowsEngine::OnLoseFocus(HWND new_focus) {
+  for (auto itr : platform_views_) {
+    HWND hwnd = itr.second;
+    if (hwnd == new_focus) {
+      auto id = itr.first;
+      FML_LOG(ERROR) << "Sending viewFocused for " << id;
+      platform_view_channel_->InvokeMethod(
+          "viewFocused", std::make_unique<EncodableValue>(id));
+      return;
+    }
+  }
+}
+
+void FlutterWindowsEngine::SendTabOut(HWND pv, int reason) {
```

Comment:
This method invokes a method on the framework to allow for directional keyboard navigation from the native view to the widget tree.